v3.0_roadmap.adoc

```diff
@@ -0,0 +1,233 @@
+= Road map for James 3.0
+
+This document aims at dividing existing and upcoming code into the following categories :
+
+ - supported
+ - experimental
+ - unsupported
+
+== Mailbox section
+
+.Feature matrix : mailbox backend
+|===
+|Feature name |Status | Comment
+
+|JPA
+|supported
+|Already existing code
+
+|Memory
+|supported
+|Reference for testing
+
+|Cassandra
+|experimental
+|Essentially we do not know about how performance will behave
+
+|ElasticSearch
+|experimental
+|
+
+|HBase
+|unsupported
+|
+
+|JCR
+|unsupported
+|
+
+|Lucene
+|unsupported
+|
+
+|Caching
+|unsupported
+|
+
+|Maildir
+|unsupported
+|Known issue on MailboxPath handling
+
+|===
+
+What being "supported implies for a backend ?
+
+ - Being MPT tested
+ - Deployement tests part of unit tests (ex : cassandra-guice)
+ - Having full support in server/data/data-@backend@
+ - Having its mappers passing generic unit tests for mappers
+ - It should at least support USER_FLAGS and NAMESPACE features on MPT tests
```

Comment:
- be able to handle a constant gatling workload for at least 4 hours with constant latency
- be tested with thunderbird (we should write scenari for that)